test_conformance/basic/test_explicit_s2v.cpp

```diff
@@ -20,131 +20,122 @@
-DECLARE_EMPTY, \
-DECLARE_EMPTY    \
-}
+/* Note: the next four arrays all must match in order and size to the
+ * ExplicitTypes enum in conversions.h!!! */
+
+#define DECLARE_S2V_IDENT_KERNELS_SET(srctype)                                 \
+    {                                                                          \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, char),                             \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, uchar),                            \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, short),                            \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, ushort),                           \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, int),                              \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, uint),                             \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, long),                             \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, ulong),                            \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, float),                            \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, half),                             \
+        DECLARE_S2V_IDENT_KERNELS(#srctype, double)                            \
+    }
 
+#define DECLARE_EMPTY_SET                                                      \
+    {                                                                          \
+        DECLARE_EMPTY, DECLARE_EMPTY, DECLARE_EMPTY, DECLARE_EMPTY,            \
+        DECLARE_EMPTY, DECLARE_EMPTY, DECLARE_EMPTY, DECLARE_EMPTY,            \
+        DECLARE_EMPTY, DECLARE_EMPTY, DECLARE_EMPTY, DECLARE_EMPTY,            \
+        DECLARE_EMPTY, DECLARE_EMPTY, DECLARE_EMPTY                            \
+    }
+
+#define NUM_VEC_TYPES 11
 
 /* The overall array */
-const char * kernel_explicit_s2v_set[kNumExplicitTypes][kNumExplicitTypes][5] = {
-    DECLARE_S2V_IDENT_KERNELS_SET(bool),
+const char * kernel_explicit_s2v_set[NUM_VEC_TYPES][NUM_VEC_TYPES][5] = {
     DECLARE_S2V_IDENT_KERNELS_SET(char),
     DECLARE_S2V_IDENT_KERNELS_SET(uchar),
-    DECLARE_S2V_IDENT_KERNELS_SET(unsigned char),
     DECLARE_S2V_IDENT_KERNELS_SET(short),
     DECLARE_S2V_IDENT_KERNELS_SET(ushort),
-    DECLARE_S2V_IDENT_KERNELS_SET(unsigned short),
     DECLARE_S2V_IDENT_KERNELS_SET(int),
     DECLARE_S2V_IDENT_KERNELS_SET(uint),
-    DECLARE_S2V_IDENT_KERNELS_SET(unsigned int),
     DECLARE_S2V_IDENT_KERNELS_SET(long),
     DECLARE_S2V_IDENT_KERNELS_SET(ulong),
-    DECLARE_S2V_IDENT_KERNELS_SET(unsigned long),
     DECLARE_S2V_IDENT_KERNELS_SET(float),
-    DECLARE_EMPTY_SET
+    DECLARE_S2V_IDENT_KERNELS_SET(half),
+    DECLARE_S2V_IDENT_KERNELS_SET(double)
 };
 
-int test_explicit_s2v_function(cl_device_id deviceID, cl_context context, cl_command_queue queue, const char *programSrc,
-                               ExplicitType srcType, unsigned int count, ExplicitType destType, unsigned int vecSize, void *inputData )
+// clang-format on
+
+int IsFloatNaN(double x)
```

Comment:
This seems a little odd and error prone since the name of the function is `IsFloatNaN`, but it's passed a `double`, then it immediately casts and assigns it to a `cl_float`.

Could we just use `std::isnan` instead - [link](https://en.cppreference.com/w/cpp/numeric/math/isnan)?